src/main.rs

```diff
@@ -1,4 +1,14 @@
-#[tokio::main]
-pub async fn main() -> miette::Result<()> {
-    pixi::cli::execute().await
+pub fn main() -> miette::Result<()> {
+    let runtime = tokio::runtime::Builder::new_current_thread()
+        .enable_all()
+        .build()
+        .expect("Failed building the Runtime");
+
+    // Box the large main future to avoid stack overflows.
+    let result = runtime.block_on(Box::pin(pixi::cli::execute()));
+
+    // Avoid waiting for pending tasks to complete.
+    runtime.shutdown_background();
+
+    result
 }
```

Comment:
Could you please open a separate PR for this change?

Reply:
See: https://github.com/prefix-dev/pixi/pull/2892